examples/platform/esp32/mode-support/static-supported-modes-manager.cpp

```diff
@@ -0,0 +1,132 @@
+#include "static-supported-modes-manager.h"
+#include "ESP32Config.h"
+#include <app/util/debug-printing.h>
+#include <app/util/ember-print.h>
+
+using namespace chip;
+using namespace chip::app::Clusters;
+using namespace chip::DeviceLayer::Internal;
+using namespace chip::app::Clusters::ModeSelect;
+using chip::Protocols::InteractionModel::Status;
+
+using ModeOptionStructType = Structs::ModeOptionStruct::Type;
+using SemanticTag          = Structs::SemanticTagStruct::Type;
+template <typename T>
+using List                                                                                = app::DataModel::List<T>;
+char chip::app::Clusters::ModeSelect::StaticSupportedModesManager::supportedModeLabel[64] = "";
+
+const StaticSupportedModesManager StaticSupportedModesManager::instance = StaticSupportedModesManager();
+
+SupportedModesManager::ModeOptionsProvider StaticSupportedModesManager::getModeOptionsProvider(EndpointId endpointId) const
+{
+    char keyBuf[ESP32Config::kMaxConfigKeyNameLength];
+    uint32_t supportedModeCount = 0;
+
+    VerifyOrReturnValue(ESP32Config::KeyAllocator::SupportedModesCount(keyBuf, sizeof(keyBuf), endpointId) == CHIP_NO_ERROR,
+                        ModeOptionsProvider(nullptr, nullptr));
+    ESP32Config::Key countKey(ESP32Config::kConfigNamespace_ChipFactory, keyBuf);
+    VerifyOrReturnValue(ESP32Config::ReadConfigValue(countKey, supportedModeCount) == CHIP_NO_ERROR,
+                        ModeOptionsProvider(nullptr, nullptr));
+
+    ModeOptionStructType * modeOptionStruct = new ModeOptionStructType[supportedModeCount + 1];
```

Comment:
Where does this memory allocation get freed?

Reply:
Freed the allocated memory in destructor.